apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -1,311 +1,479 @@
+                            received_time: Some(span.end_time.into()),
+                        }),
+                    )),
+                })]
+            }
+            "flatten" => vec![TreeData::QueryPlanNode(QueryPlanNode {
+                node: Some(apollo_spaceport::trace::query_plan_node::Node::Flatten(
+                    Box::new(FlattenNode {
+                        response_path: vec![],
+                        node: None,
+                    }),
+                )),
+            })],
+            "subgraph" => {
+                vec![TreeData::Trace(self.find_ftv1_trace(span))]
+            }
+            _ => child_nodes,
+        })
     }
 
-    // XXX CANNOT USE SIMPLE WITH OUR IMPLEMENTATION AS NO RUNTIME EXISTS
-    // WHEN TRYING TO EXPORT...
-    /// Install the apollo telemetry exporter pipeline with the recommended defaults.
-    #[allow(dead_code)]
-    pub(crate) fn install_simple(mut self) -> Result<sdk::trace::Tracer, ApolloError> {
-        let exporter = self.build_exporter()?;
-
-        let mut provider_builder =
-            sdk::trace::TracerProvider::builder().with_simple_exporter(exporter);
-        if let Some(config) = self.trace_config.take() {
-            provider_builder = provider_builder.with_config(config);
+    fn find_ftv1_trace(
+        &mut self,
+        span: &SpanData,
+    ) -> Result<Option<Box<apollo_spaceport::Trace>>, Error> {
+        if !self.field_level_instrumentation {
+            Ok(None)
+        } else {
+            Ok(span
+                .attributes
+                .get(&Key::new("apollo_private_ftv1"))
+                .map(|data| {
+                    if let Value::String(data) = data {
+                        Ok(Box::new(apollo_spaceport::Trace::decode(Cursor::new(
+                            base64::decode(data.to_string())?,
+                        ))?))
+                    } else {
+                        Err(Error::Ftv1SpanAttribute)
+                    }
+                })
+                .transpose()?)
         }
-        let provider = provider_builder.build();
-
-        let tracer = provider.versioned_tracer(
-            "apollo-opentelemetry",
-            Some(env!("CARGO_PKG_VERSION")),
-            None,
-        );
-        let _prev_global_provider = global::set_tracer_provider(provider);
-
-        Ok(tracer)
     }
 
-    /// Create a client to talk to our spaceport and return an exporter.
-    pub(crate) fn build_exporter(&self) -> Result<Exporter, ApolloError> {
-        let collector = match self.spaceport_config.clone() {
-            Some(cfg) => cfg.collector,
-            None => DEFAULT_SERVER_URL.to_string(),
+    fn extract_http_data(&self, span: &SpanData) -> Http {
+        let method = match span
+            .attributes
+            .get(&Key::new("method"))
+            .map(|data| data.as_str())
+            .unwrap_or_default()
+            .as_ref()
+        {
+            "OPTIONS" => 1,
```

Comment:
It looks like you can do `apollo_spaceport::trace::http::Method::Options as i32` etc here, so that while you probably still have to list all the methods explicitly, you at least don't have to ensure that the numbers match the proto file.

Reply:
Fixed in [28e2b87](https://github.com/apollographql/router/pull/1514/commits/28e2b8701701485c1bfcaf7b9d087d24954d3629)